lib/deimos/utils/db_poller/base.rb

```diff
@@ -21,17 +21,39 @@ class Base
         # @return [Hash]
         attr_reader :config
 
+        # Method to define producers if a single poller needs to publish to multiple topics.
+        # Producer classes should be constantized
+        # @return [Array<Producer>]
+        def self.multi_producers
+          []
+        end
+
         # @param config [FigTree::ConfigStruct]
         def initialize(config)
           @config = config
           @id = SecureRandom.hex
+          @producers = []
           begin
-            @producer = @config.producer_class.constantize
-          rescue NameError
-            raise "Class #{@config.producer_class} not found!"
-          end
-          unless @producer < Deimos::ActiveRecordProducer
-            raise "Class #{@producer.class.name} is not an ActiveRecordProducer!"
+            @poller_class = @config.poller_class.present? ? @config.poller_class.constantize : nil
+            if self.class.multi_producers.empty?
+              if @config.producer_class.nil?
+                raise 'No producers have been set for this DB poller!'
+              end
+
+              @producer = @config.producer_class.constantize
+              @producers << @producer
+              unless @producer < Deimos::ActiveRecordProducer
+                raise "Class #{@producer.class.name} is not an ActiveRecordProducer!"
+              end
+            else
+              @producers.append(self.class.multi_producers).flatten!.each do |producer|
```

Comment:
Is there a reason we need to flatten it? And why would we need to append the producer list again when we already have it?

Reply:
@dorner The reason why I appended the producer list to the @producers instance variable is so that I could use the @producers instance variable throughout the base file (as I set it in both the multi producer and single producer configurations), especially when doing the Deimos config resource tracer spans.

Reply:
But the point is that either `self.producers` returns a value, or `@producer` is given. There's no reason to combine the two since only one of them will have a value.